go/vt/binlog/event_streamer.go

```diff
@@ -194,34 +200,93 @@ func parsePkTuple(tokenizer *sqlparser.Tokenizer, insertid int64) ([]interface{}
 			if typ2 != sqlparser.NUMBER {
 				return nil, insertid, fmt.Errorf("expecting number after '-'")
 			}
-			valstr := string(val2)
-			if ival, err := strconv.ParseInt(valstr, 0, 64); err == nil {
-				result = append(result, -ival)
-			} else {
+
+			// check value
+			fullVal := append([]byte{'-'}, val2...)
+			if _, err := strconv.ParseInt(string(fullVal), 0, 64); err != nil {
 				return nil, insertid, err
 			}
+
+			// update type
+			switch fields[index].Type {
+			case mproto.VT_DECIMAL:
```

Comment:
It's ironic that VT_DECIMAL is 0, and VT_NULL is 6...
The code would have been so less confusing if VT_NULL were 0.